### DISCUSSION TITLE
telemetry and lap number

### DISCUSSION THREAD

**Discussion by McHouse** [theOehrly/Fast-F1#511] [Q&A] [answered]
Hello and thanks to those who spend their time on these F1 matters.

I am using this:

```
drivers_ini = results["Abbreviation"]
drivers_num = results["DriverNumber"]
drivers_nam = results["BroadcastName"]

time0 = time.time()
n = 0
race = {}
for car in (drivers_num):
    df = laps.pick_driver(int(car)).telemetry
    df['Driver'] = car
    initials = drivers_ini[n]
    race[initials] = df
    print(initials)
    n += 1

print(time.time()-time0)
```

Is there a way to indicate which lap each telemetry record (or group of records) belongs to?

Thanks in advance.

**Comment by theOehrly** [comment, marked as answer]
No, there isn't. You would need to do this manually by comparing the timestamps with the timestamps of a drivers' laps.

Alternatively, you could iterate over all laps and get the telemetry for each lap individually.

**Comment by McHouse** [comment]
Thank you very much theOehrly, I was already afraid of that, but as a beginner that I am...
Iterating by laps is unfeasible because of how slow it is.
I guess filtering by timestamps won't be much faster.

A shame.
Thank you very much, especially for the speed.👍

**Reply by theOehrly** [reply]
Iterating over the timestamps shouldn't take that long. I would expect a second or two at the worst.

**Comment by McHouse** [comment]
TheOherly, sorry for my clumsiness, but...

first: It's not a shame, it's just that I'm already embarrassed to ask newbie things. That's because of my final comment in the previous message.
second: For the same reason as before, thank you for the correct orientation to my problems. I was very and pleasantly surprised by how little it took to filter the multitude of telemetry records of a GP.
12 seconds to generate race telemetries and 1.8 to filter them. Awesome.
Thousands of thanks.
